File: backend/src/main/java/com/clinicmanagement/user/UserService.java
package com.clinicmanagement.user;

import com.clinicmanagement.user.dto.CreateUserRequest;
import com.clinicmanagement.user.dto.UpdateUserRequest;
import com.clinicmanagement.user.dto.UserSummaryResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

public interface UserService {

    Page<UserSummaryResponse> getUsers(String keyword, String status, String role, Pageable pageable);

    UserSummaryResponse getUserById(Long userId);

    UserSummaryResponse createUser(CreateUserRequest request);

    UserSummaryResponse updateUser(Long userId, UpdateUserRequest request);

    UserSummaryResponse updateCurrentUser(User currentUser, UpdateUserRequest request);

    UserSummaryResponse uploadCurrentUserAvatar(User currentUser, MultipartFile file);

    void lockUser(Long userId);

    void unlockUser(Long userId);

    void deleteUser(Long userId);
}

File: frontend/src/services/userService.js
import axiosClient from "./axiosClient";

export const getUsers = (params, config = {}) => axiosClient.get("/users", { params, ...config });
export const createUser = (payload) => axiosClient.post("/users", payload);
export const updateUser = (userId, payload) => axiosClient.put(`/users/${userId}`, payload);
export const updateCurrentUser = (payload) => axiosClient.put("/users/me", payload);
export const uploadCurrentUserAvatar = (file) => {
  const formData = new FormData();
  formData.append("file", file);
  return axiosClient.post("/users/me/avatar", formData, {
    headers: { "Content-Type": "multipart/form-data" },
  });
};
export const lockUser = (userId) => axiosClient.put(`/users/${userId}/lock`);
export const unlockUser = (userId) => axiosClient.put(`/users/${userId}/unlock`);
export const deleteUser = (userId) => axiosClient.delete(`/users/${userId}`);
